test/org/sagebionetworks/bridge/services/ActivityEventServiceTest.java

```diff
@@ -225,6 +233,131 @@ public void canPublishEnrollmentEventWithAutomaticCustomEvents() {
                 .getTimestamp().longValue());
         assertEquals("AAA-BBB-CCC", publishedEventList.get(3).getHealthCode());
     }
+    
+    @Test
+    public void whenNoEnrollmentEventPublishNoCustomEvents() {
+        // Configure study with automatic custom events
+        Study study = Study.create();
+        // Note that these events include events that are implicitly and explicitly related to 
+        // enrollment, and some that are not applicable that should be ignored.
+        study.setAutomaticCustomEvents(ImmutableMap.<String, String>builder()
+                .put("3-days-after", "P3D") // defaults to enrollment
+                .put("1-week-after", "enrollment:P1W")
+                .put("13-weeks-after", "enrollment:P13W")
+                .put("5-years-after", "not_enrollment:P5Y")
+                .put("10-years-after", "not_entrollment:P10Y").build());
+        
+        when(activityEventDao.publishEvent(any())).thenReturn(false);
+        
+        activityEventService.publishEnrollmentEvent(study,"AAA-BBB-CCC", new ConsentSignature.Builder().build());
+        
+        // Only happens once, none of the other custom events are published.
+        verify(activityEventDao, times(1)).publishEvent(any());
+    }
+    
+    @Test
+    public void whenNoActivitiesRetrievedEventPublishNoCustomEvents() {
+        // Configure study with automatic custom events
```

Comment:
Is it worth adding an equivalent test for custom events publishing automatic custom events?

Reply:
Added similar tests that automatic custom events don't fire for enrollment or custom events, if the DAO returns false.